### DISCUSSION TITLE
How to use the AzureOpenAI API with the wizard?

### DISCUSSION THREAD

**Discussion by rosch100** [AliDehbansiahkarbon/ChatGPTWizard#53] [Q&A]
The URL for using the AzureOpenAI API is:
` https://YOUR_RESOURCE_NAME.openai.azure.com/openai/deployments/YOUR_DEPLOYMENT_NAME/completions?api-version=2023-05-15` for GPT-3
and:
` https://YOUR_RESOURCE_NAME.openai.azure.com/openai/deployments/YOUR_DEPLOYMENT_NAME/chat/completions?api-version=2023-05-15`
 for GPT-4 and GPT-3.5 Turbo.
(see: [Azure OpenAI reference](https://learn.microsoft.com/en-us/azure/ai-services/openai/reference))

I can access this API e.g:
```
curl 'https:///YOUR_RESOURCE_NAME.openai.azure.com/openai/deployments/YOUR_DEPLOYMENT_NAME/chat/completions?api-version=2023-05-15' \
  -H "Content-Type: application/json" \
  -H "api-key: YOUR_API_KEY" \
  -d "{
   \"messages\": [
      {
        \"role\": \"system\", \"content\": \"Assistant is a large language model trained by OpenAI.\"
      },
     {
        \"role\": \"user\", \"content\": \"What is the difference between garbanzo beans and chickpeas?\"
     }
   ]
}"
```
However, I could not use the Azure completions URL with ChatGPTWizard. Using the above URL with my API key, I get the response:
`No answer, try again`

I suspect this to be a malformed API call with the above URL.

**Comment by AliDehbansiahkarbon** [comment]
Thank you for providing sufficient information about the issue. While this plugin was not originally designed to support AzureOpenAI, I believe it should still function with some minor modifications. I will make the necessary changes and update you on the progress.

**Comment by fabiohmribeiro** [comment]
Has there been any update in this regard?

**Reply by AliDehbansiahkarbon** [reply]
Hi Fabio,
Unfortunately No, not yet.
I have been working on another plug-in so far, and it's almost done.
When I finish that I will start working on CHatGPTWizard, there are more exciting updates but it takes time.
Please be aware I'm alone and pretty busy these days but I promise I will do that.
**I appreciate your patience.**👍

Best regards,
Ali.
